binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -410,7 +410,7 @@ public Map<String, Object> mergedProducerConfiguration() {
 
 	private void filterStreamManagedConfiguration(Map<String, Object> configuration) {
 		if (configuration.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)
-				&& configuration.get(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG).equals(true)) {
+				&& configuration.get(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG).equalsIgnoreCase("true")) {
```

Comment:
This doesn't look right. You are calling `eqaulsIgnoreCase` on an `Object` not on `String`. Did you build the project after this change? Also, did you verify that this resolved the errors you were seeing?

Reply:
@sobychacko Hello, the code logic has been adjusted.